sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/eventhubs/kafka/AzureEventHubsKafkaAutoConfiguration.java

```diff
@@ -3,54 +3,60 @@
 import org.springframework.kafka.core.KafkaTemplate;
+import org.springframework.util.Assert;
+
+import java.util.ArrayList;
+import java.util.Collections;
 
 
 /**
- * An auto-configuration for Event Hub, which provides {@link KafkaProperties}
+ * An auto-configuration for Azure Event Hubs Kafka, which provides {@link KafkaProperties}
  *
- * @author Warren Zhu
  */
 @Configuration(proxyBeanMethods = false)
 @ConditionalOnClass(KafkaTemplate.class)
-//@EnableConfigurationProperties(AzureEventHubProperties.class)
+@ConditionalOnProperty(value = "spring.cloud.azure.eventhubs.kafka.enabled", havingValue = "true", matchIfMissing = true)
+@AutoConfigureAfter({AzureEventHubsAutoConfiguration.class, AzureEventHubsResourceManagerAutoConfiguration.class })
 public class AzureEventHubsKafkaAutoConfiguration {
-    private static final String SECURITY_PROTOCOL = "security.protocol";
-    private static final String SASL_SSL = "SASL_SSL";
-    private static final String SASL_JAAS_CONFIG = "sasl.jaas.config";
-    private static final String SASL_CONFIG_VALUE = "org.apache.kafka.common.security.plain.PlainLoginModule required"
-        + " username=\"$ConnectionString\" " + "password=\"%s\";%n";
-    private static final String SASL_MECHANISM = "sasl.mechanism";
-    private static final String SASL_MECHANISM_PLAIN = "PLAIN";
-    private static final int PORT = 9093;
-
-    @SuppressWarnings("rawtypes")
+
+    private static final String SASL_CONFIG_VALUE = "org.apache.kafka.common.security.plain.PlainLoginModule required username=\"$ConnectionString\" password=\"%s\";%s";
+
     @Primary
     @Bean
-    // TODO (xiada): refactor this logic
-    public KafkaProperties kafkaProperties(
-        AzureEventHubsProperties eventHubsProperties,
+    @ConditionalOnBean(value = AzureServiceType.EventHubs.class, parameterizedContainer = ConnectionStringProvider.class)
+    public KafkaProperties azureKafkaProperties(
         ObjectProvider<ConnectionStringProvider<AzureServiceType.EventHubs>> connectionStringProviders) {
         KafkaProperties kafkaProperties = new KafkaProperties();
-/*
-        String endpoint = namespace.serviceBusEndpoint();
-        String endpointHost = endpoint.substring("https://".length(), endpoint.lastIndexOf(':'));
-        kafkaProperties.setBootstrapServers(Arrays.asList(endpointHost + ":" + PORT));
-        kafkaProperties.getProperties().put(SECURITY_PROTOCOL, SASL_SSL);
-        kafkaProperties.getProperties().put(SASL_MECHANISM, SASL_MECHANISM_PLAIN);
-        kafkaProperties.getProperties().put(SASL_JAAS_CONFIG,
-            String.format(SASL_CONFIG_VALUE, connectionStringProviders.getIfAvailable().getConnectionString(), System.getProperty("line.separator")));*/
+
+        ConnectionStringProvider<AzureServiceType.EventHubs> connectionStringProvider =
+            connectionStringProviders.getIfAvailable();
+        Assert.notNull(connectionStringProvider, "Connection string provider must not be null.");
```

Comment:
if this is required, then we should not use _ObjectProvider_ as the input argument